### PR TITLE
USHIFT-1977: add strictfipsruntime tag to microshift binary build

### PR DESCRIPTION
**Pull request by eslutsky**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -102,7 +102,7 @@ debug:
 	@echo PATCH:"$(PATCH)"
 
 # These tags make sure we can statically link and avoid shared dependencies
```
Comment (dhellmann) [inline Makefile:104]: This comment is no longer accurate, we should have removed it before when we enabled cgo.

**Review by dhellmann — COMMENTED** [review]
/lgtm
/hold

One nit, but nothing worth holding up the PR. I'll leave a hold in place in case there are test issues. Feel free to remove it when you're ready to merge.

**Comment by eslutsky** [comment]
it appears that those build flags only works with podman build  (doesnt work with upstream golang builds)

File: Makefile [hunk 2]
```diff
@@ -228,6 +228,15 @@ validate-cluster:
 OS := $(shell go env GOOS)
 ARCH := $(shell go env GOARCH)
 
+
+# check if FIPS supported
```
[the remark this answers is not in the document]
Comment (eslutsky) [reply Makefile:232]: the only place that its officially documented its here https://docs.google.com/presentation/d/1o3IowxHX6BsnxGkIInaQ0lBgnn_K5Ex8jxwCYCeNsqs/edit#slide=id.g2679cb578c3_0_17

File: Makefile (hunk 1 above)
Comment (eslutsky) [reply Makefile:104]: we should address this in separate PR

**Comment by dhellmann** [comment]
/override ci/prow/microshift-metal-tests

The failing test (`el92-src@storage.Reboot: Rebooting Healthy System Should Keep Functional PVC `) is not related to this change.

**Pull request merged**